src/js/numbers.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * NUMBER OPERATIONS
+ */
+
+/* global alert, location, jQuery, localStorage, navigator */
+import { PRIMES } from './constants.js'
+import { get_prime_factors } from './sequences.js'
+
+// modulo function
+Number.prototype.mod = function (n) {
+  return ((this % n) + n) % n;
+};
+
+// calculate the sum of the values in a given array given a stopping index
+function sum_array(array, index) {
+  var sum = 0;
+
+  if (array.length <= index)
+    index = array.length - 1;
+
+  for (let i = 0; i < index; i ++) {
+    sum += array[i];
+  }
+
+   return sum;
+}
+
+// rotates the array by given steps
+function rotate(array, steps) {
```

Comment:
Note to myself: need to create an immutable version of this function

Reply:
I could do that too if you want. Would you just name it differently or add a bool arg?